<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tabStepper">
       <attribute name="title">
        <string>stepper</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="2">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>baud:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QPushButton" name="pushButtonConnectSerialport">
          <property name="text">
           <string>connect</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QComboBox" name="comboBoxBaud">
          <item>
           <property name="text">
            <string>4800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>14400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>28800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>38400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>57600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>115200</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>port:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEditSerialport">
          <property name="text">
           <string>/dev/ttyACM0</string>
          </property>
         </widget>
        </item>
        <item row="3" column="4">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="4">
         <widget class="QPushButton" name="pushButtonTestSerialport">
          <property name="text">
           <string>test</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" rowspan="3" colspan="4">
         <widget class="QTextBrowser" name="textBrowserSerialport"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabCam">
       <attribute name="title">
        <string>cam</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3" rowstretch="0,0,0,0,0,0,0" columnstretch="0,0,1,0,0,0" columnminimumwidth="0,0,0,0,0,0">
        <property name="leftMargin">
         <number>6</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="labelCam">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>cam:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QPushButton" name="pushButtonCamShow">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>show</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelFPS">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>fps:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBoxCamFPS">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>25</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QPushButton" name="pushButtonCamClose">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>close</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="2">
         <widget class="QPushButton" name="pushButtonCamCalibration">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>calibration</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0" colspan="2">
         <widget class="QPushButton" name="pushButtonCamCalibrationCollect">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>collect</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="comboBoxCam">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <item>
           <property name="text">
            <string>0</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>3</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>5</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="0">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2" rowspan="7">
         <widget class="Cam_viewer" name="graphicsViewCam"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabModel">
       <attribute name="title">
        <string>model</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>400</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QToolBar" name="toolBar_3">
   <property name="windowTitle">
    <string>toolBar_3</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Cam_viewer</class>
   <extends>QGraphicsView</extends>
   <header>cam_viewer.h</header>
   <slots>
    <slot>change_fps(int)</slot>
    <slot>cam_show()</slot>
    <slot>cam_close()</slot>
    <slot>cam_calibration()</slot>
    <slot>cam_calibration_collect()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButtonTestSerialport</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>test_serialport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>339</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonConnectSerialport</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>connect_serialport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>339</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCamShow</sender>
   <signal>clicked()</signal>
   <receiver>graphicsViewCam</receiver>
   <slot>cam_show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>339</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCamClose</sender>
   <signal>clicked()</signal>
   <receiver>graphicsViewCam</receiver>
   <slot>cam_close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>337</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCamCalibration</sender>
   <signal>clicked()</signal>
   <receiver>graphicsViewCam</receiver>
   <slot>cam_calibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>337</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonCamCalibrationCollect</sender>
   <signal>clicked()</signal>
   <receiver>graphicsViewCam</receiver>
   <slot>cam_calibration_collect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>337</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBoxCamFPS</sender>
   <signal>valueChanged(int)</signal>
   <receiver>graphicsViewCam</receiver>
   <slot>change_fps(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>connect_serialport()</slot>
  <slot>test_serialport()</slot>
 </slots>
</ui>
